### PR TITLE
use canonical filename of RTD configuration and enable `nitpicky` mode

### PR DESCRIPTION
**Pull request by zacharyburnett**
### Description

update the name of the ReadTheDocs configuration file to `.readthedocs.yaml` and enable `nitpicky` mode to validate intra-doc links

### CODE REVIEW

**Comment by pllim** [comment]
Please also fix the 3 warnings. Thanks!

**Comment by zacharyburnett** [comment]
for reference, warnings are https://readthedocs.org/projects/stginga/builds/18123433/:
```
/home/docs/checkouts/readthedocs.org/user_builds/stginga/envs/226/lib/python3.9/site-packages/stginga/utils.py:docstring of stginga.utils.scale_image:16: WARNING: py:func reference target not found: scipy.ndimage.interpolation.zoom
/home/docs/checkouts/readthedocs.org/user_builds/stginga/envs/226/lib/python3.9/site-packages/stginga/utils.py:docstring of stginga.utils.scale_image_with_dq:16: WARNING: py:func reference target not found: scipy.ndimage.interpolation.zoom
/home/docs/checkouts/readthedocs.org/user_builds/stginga/envs/226/lib/python3.9/site-packages/stginga/utils.py:docstring of stginga.utils.scale_wcs:21: WARNING: py:func reference target not found: scipy.ndimage.interpolation.zoom
```

File: docs/conf.py [hunk 1]
```diff
@@ -121,10 +122,16 @@
...
 # -- Options for linkcheck output ---------------------------------------------
 linkcheck_retry = 5
 linkcheck_ignore = ['https://hsthelp.stsci.edu', 'https://www.ball.com']
 linkcheck_timeout = 180
 linkcheck_anchors = True
+
+# Enable nitpicky mode - which ensures that all references in the docs resolve.
+nitpicky = True
+nitpick_ignore = [
+    ('py:func', 'scipy.ndimage.interpolation.zoom'),
```
Comment (pllim) [inline docs/conf.py]: Why not just change it to `scipy.ndimage.zoom` in the docs?

https://docs.scipy.org/doc/scipy/reference/generated/scipy.ndimage.zoom.html

Comment (zacharyburnett) [reply docs/conf.py]: oh good idea, sorry I don't work with `scipy` often so when I do it's difficult to keep track of where everything is

**Review by pllim — APPROVED** [review]
Thanks!

**Pull request merged**